packages/generators/app/lib/utils/check-requirements.js

```diff
@@ -1,14 +1,28 @@
 'use strict';
 
+const { red, green, bold, yellow } = require('chalk');
+const semver = require('semver');
+const packageJSON = require('../resources/json/common/package.json');
+
 module.exports = function checkBeforeInstall() {
   const currentNodeVersion = process.versions.node;
-  const semver = currentNodeVersion.split('.');
-  const major = semver[0];
+  const { engines } = packageJSON({ strapiDependencies: [] });
 
-  if (major < 12) {
-    console.error(`You are running Node ${currentNodeVersion}`);
-    console.error('Strapi requires Node 12 and higher.');
+  // error if the node version isn't supported
+  if (!semver.satisfies(currentNodeVersion, engines.node)) {
+    console.error(red(`You are running ${bold(`Node.js ${currentNodeVersion}`)}`));
+    console.error(`Strapi requires ${bold(green(`Node.js ${engines.node}`))}`);
     console.error('Please make sure to use the right version of Node.');
     process.exit(1);
   }
+
+  // warn if not using a LTS version
+  else if (semver.satisfies(currentNodeVersion, '15.x.x || 17.x.x || 19.x.x')) {
```

Comment:
@derrickmehaffy Regarding your comments about not allowing node versions that aren't supported, I've updated that, but had another question.

What do you think about adding this warning for non-LTS versions? Previously we allowed it without warning. We could also block those as well if you think it's better.